mlxtend/association/association_rules.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+import pandas as pd
+from itertools import combinations
+
+def association_rules(df, metric="confidence", min_threshold=0.8):
+    """"""
+    # Metrics for association rules
+    metric_dict = {
+        "confidence": lambda sXY, sX, _: \
+        sXY/sX,
+        "lift": lambda sXY, sX, sY:\
+        metric_dict["confidence"](sXY, sX, sY)/sY
+        }
+    
+    # check for metric compliance
+    if metric not in metric_dict.keys():
+        raise ValueError("Metric must be 'confidence' or 'lift', got '{}'".format(metric))
+        
+    # get dict of {frequent itemset} -> support
+    frozenset_vect = np.vectorize(lambda x: frozenset(x))
+    keys = df.values.T[1]
+    values = df.values.T[0]
+    frequent_items_dict = dict(zip(frozenset_vect(keys), values))
+    
+    # prepare buckets to collect frequent rules
+    rule_antecedants = []
+    rule_consequents = []
+    rule_supports = []
+    
+    # iterate over all frequent itemsets
+    for k in frequent_items_dict.keys():
+        sXY = frequent_items_dict[k]
+        # to find all possible combinations
+        for idx in range(len(k)-1, 0, -1):
+            # of antecedent and consequent
+            for c in combinations(k, r=idx):
+                antecedent = frozenset(c)
+                consequent = k.difference(antecedent)
+                sX = frequent_items_dict[antecedent]
+                sY = frequent_items_dict[consequent]
+                # check for the threshold
+                if metric_dict[metric](sXY, sX, sY) >= min_threshold:
+                    rule_antecedants.append(antecedent)
+                    rule_consequents.append(consequent)
+                    rule_supports.append([sXY, sX, sY])
+                    
+    # check if frequent rule was generated
+    if not rule_supports:
+        return
```

Comment:
I'd say it's better to always return a DataFrame with the same column structure (makes unit testing and other things maybe easier). E.g., something like `pd.DataFrame(columns=["antecedants", "consequents", "support"] + list(metric_dict.keys()))`